interfaces/smtp/v0/schema.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 Canonical
+# See LICENSE file for licensing details.
+"""This file defines the schema for the provider side of the smtp interface.
+
+It exposes one interfaces.schema_base.DataBagSchema subclass called:
+- ProviderSchema
+
+Examples:
+    ProviderSchema:
+        unit: <empty>
+        app: {"smtp":
+                 {
+                    "host": "example.smtp",
+                    "port": "25",
+                    "user": "example_user",
+                    "password_id": "01548499c9233d4612352c989162d940f6a9e6f6d5cc058dfcf66f51575e09c2",
+                    "auth_type": "plain",
+                    "transport_security": "tls",
+                    "domain": "domain",
+                }
+             }
+"""
+from enum import Enum
+from interface_tester.schema_base import DataBagSchema
+from pydantic import BaseModel, Field
+from typing import Optional
+
+
+class TransportSecurity(str, Enum):
+    """Represent the transport security values.
+
+    Attributes:
+        NONE: none
+        STARTTLS: starttls
+        TLS: tls
+    """
+
+    NONE = "none"
+    STARTTLS = "starttls"
+    TLS = "tls"
+
+
+class AuthType(str, Enum):
+    """Represent the auth type values.
+
+    Attributes:
+        NONE: none
+        NOT_PROVIDED: not_provided
+        PLAIN: plain
+    """
+
+    NONE = "none"
+    NOT_PROVIDED = "not_provided"
+    PLAIN = "plain"
+
+
+class SmtpProviderData(BaseModel):
+    host: str = Field(
+        ...,
+        min_length=1,
+        description="SMTP host.",
+        title="Host",
+        examples=["example.smtp"],
+    )
+    port: int = Field(
+        None,
+        ge=1,
+        le=65536,
+        description="SMTP port.",
+        title="Port",
+        examples=[25],
+    )
+    user: Optional[str] = Field(
+        description="SMTP user.",
+        title="User",
+        examples=["some_user"],
+    )
+    password: Optional[str] = Field(
+        description="SMTP password.",
+        title="Password",
+        examples=["somepasswd"],
+    )
+    password_id: Optional[str] = Field(
+        description="Juju secret ID for the SMTP password.",
+        title="Password ID",
+        examples=["secret:123213123123123123123"],
+    )
+    auth_type: AuthType = Field(
+        description="The type used to authenticate with the SMTP relay.",
+        title="Auth type",
+        examples=[AuthType.NONE],
+    )
+    transport_security: TransportSecurity = Field(
+        description="The security protocol to use for the outgoing SMTP relay.",
+        title="Transport security",
+        examples=[TransportSecurity.NONE],
+    )
+    domain: Optional[str] = Field(
+        description="The domain used by the sent emails from SMTP relay.",
```

Comment:
'domain' is an overloaded term in email. I assume here this is the enveloper sender domain? (Sometimes called "MAIL FROM" domain). If so, could that be more explicit? Or is this meant to be either envelope sender or envelope recipient depending on whether the smtp provider is accepting mail for ~local delivery or accepting mail to send out to the world? I assume it's not any header domain, though? Or maybe the DKIM/DMARC domain? Whatever it is, it would help if this was clearer.

Reply:
Yes, that's correct. By explicit, you mean renaming the field or does expanding the description suffice?

Reply:
I think expanding the description would be fine.